### PR TITLE
NERT-565: Initial codes page for admin editing

### PR DESCRIPTION
**Pull request by KjartanE**
## Links to Jira Tickets

- https://apps.nrs.gov.bc.ca/int/jira/browse/NERT-565

## Description of Changes

- {List all relevant code changes. Include any changes to the business workflow that might not be obvious to the reviewers of this PR.}

## Testing Notes

- {List any relevant testing considerations, necessary pre-reqs, and areas of the app to focus on. Specifically, include anything that will help the reviewers of this PR verify the code is functioning as expected.}

### CODE REVIEW

**Comment by oscar-ip** [comment]
Looks like there is an issue, I am getting a 500 error, when trying to save a title
"Dev Restoration tracker hello world sadf SD ASD AS D as a S as A S as S  as  Sa s  S as A S  s AS a s S  s"

https://restoration-tracker-api-253-d83219-dev.apps.silver.devops.gov.bc.ca/api/codes/
{name: "Internal Server Error", status: 500, message: "Unexpected Error",…}
errors
: 
["Error executing SQL query", "Failed to execute SQL"]
message
: 
"Unexpected Error"
name
: 
"Internal Server Error"
status
: 
500

**Comment by oscar-ip** [comment]
Additionally, the UI needs a bit of work
when you Edit the title and after if you edit the email the form is prepopulated with whatever you entered in the title
image

**Comment by KjartanE** [comment]
> Looks like there is an issue, I am getting a 500 error, when trying to save a title "Dev Restoration tracker hello world sadf SD ASD AS D as a S as A S as S as Sa s S as A S s AS a s S s"
> 
> https://restoration-tracker-api-253-d83219-dev.apps.silver.devops.gov.bc.ca/api/codes/ {name: "Internal Server Error", status: 500, message: "Unexpected Error",…} errors : ["Error executing SQL query", "Failed to execute SQL"] message : "Unexpected Error" name : "Internal Server Error" status : 500

Resolved with limiting text field input length

**Comment by KjartanE** [comment]
> Additionally, the UI needs a bit of work when you Edit the title and after if you edit the email the form is prepopulated with whatever you entered in the title image

fixxed with clearing input on close

**Comment by oscar-ip** [comment]
One more thing, could you also validate the input for an email address?

File: app/src/features/admin/codes/CodeDialog.tsx [hunk 1]
```diff
@@ -50,16 +61,24 @@ const CodeDialog: React.FC<CodeDialogProps> = ({
     onClose();
   };
 
+  const validateEmail = (email: string) => {
+    // don't remember from where i copied this code, but this works.
```
Comment (popkinj) [inline app/src/features/admin/codes/CodeDialog.tsx:65]: Love this regex... nice work.

**Review by popkinj — APPROVED** [review]
Great work!

**Pull request merged**